src/blueapi/config.py

```diff
@@ -89,83 +90,16 @@ class ApplicationConfig(BlueapiBaseModel):
     api: RestConfig = Field(default_factory=RestConfig)
     scratch: ScratchConfig | None = None
 
-    def __eq__(self, other: object) -> bool:
-        if isinstance(other, ApplicationConfig):
-            return (
-                (self.stomp == other.stomp)
-                & (self.env == other.env)
-                & (self.logging == other.logging)
-                & (self.api == other.api)
-            )
-        return False
-
-
-C = TypeVar("C", bound=BaseModel)
-
-
-class ConfigLoader(Generic[C]):
-    """
-    Small utility class for loading config from various sources.
-    You must define a config schema as a dataclass (or series of
-    nested dataclasses) that can then be loaded from some combination
-    of default values, dictionaries, YAML/JSON files etc.
-    """
-
-    def __init__(self, schema: type[C]) -> None:
-        self._adapter = TypeAdapter(schema)
-        self._values: dict[str, Any] = {}
-
-    def use_values(self, values: Mapping[str, Any]) -> None:
-        """
-        Use all values provided in the config, override any defaults
-        and values set by previous calls into this class.
-
-        Args:
-            values (Mapping[str, Any]): Dictionary of override values,
-                                        does not need to be exhaustive
-                                        if defaults provided.
-        """
-
-        def recursively_update_map(old: dict[str, Any], new: Mapping[str, Any]) -> None:
-            for key in new:
-                if (
-                    key in old
-                    and isinstance(old[key], dict)
-                    and isinstance(new[key], dict)
-                ):
-                    recursively_update_map(old[key], new[key])
-                else:
-                    old[key] = new[key]
-
-        recursively_update_map(self._values, values)
-
-    def use_values_from_yaml(self, path: Path) -> None:
-        """
-        Use all values provided in a YAML/JSON file in the
-        config, override any defaults and values set by
-        previous calls into this class.
-
-        Args:
-            path (Path): Path to YAML/JSON file
-        """
-
-        with path.open("r") as stream:
-            values = yaml.load(stream, yaml.Loader)
-        self.use_values(values)
-
-    def load(self) -> C:
-        """
-        Finalize and load the config as an instance of the `schema`
-        dataclass.
-
-        Returns:
-            C: Dataclass instance holding config
-        """
-
-        try:
-            return self._adapter.validate_python(self._values)
-        except ValidationError as exc:
-            error_details = "\n".join(str(e) for e in exc.errors())
-            raise InvalidConfigError(
-                f"Something is wrong with the configuration file: \n {error_details}"
-            ) from exc
+    model_config = SettingsConfigDict(
+        env_nested_delimiter="__", yaml_file=DEFAULT_PATH, yaml_file_encoding="utf-8"
+    )
```

Comment:
Should: rather than setting this class variable then calling the constructor a constructor that takes a yaml file, or a class method?

e.g.

```python
class ApplicationConfig(...):
    def __init__(self, yaml_file: Path | tuple[Path, ...] | None = None):
        self.model_config = SettingsConfigDict(env_nested_delimiter="__", yaml_file=DEFAULT_PATH or yaml_file, yaml_file_encoding="utf-8")
        super().__init__()

    #  or 
    @classmethod
    def from_yaml(cls, yaml_file: Path | tuple[Path, ...]):
          cls.model_config = yaml_field
          return cls()
```

Reply:
This means that instantiating the Config a second time isn't modified by instantiating it the first time.

Reply:
there is some Pydantic magic behind the scenes. I agree that this setup is odd but that's what the docs use

https://docs.pydantic.dev/latest/concepts/pydantic_settings/#changing-priority

Reply:
I'm hesitant to change that as now we're using the `settings_customize_sources`

Reply:
in pydantic_settings model_config _is_ specified as a ClassVariable
```python


    model_config: ClassVar[SettingsConfigDict] = SettingsConfigDict(
        extra='forbid',
        arbitrary_types_allowed=True,
        validate_default=True,
        case_sensitive=False,
        env_prefix='',
        nested_model_default_partial_update=False,
        env_file=None,
        env_file_encoding=None,
        env_ignore_empty=False,
        env_nested_delimiter=None,
        env_parse_none_str=None,
        env_parse_enums=None,
        cli_prog_name=None,
        cli_parse_args=None,
        cli_parse_none_str=None,
        cli_hide_none_type=False,
        cli_avoid_json=False,
        cli_enforce_required=False,
        cli_use_class_docs_for_groups=False,
        cli_exit_on_error=True,
        cli_prefix='',
        cli_flag_prefix_char='-',
        cli_implicit_flags=False,
        cli_ignore_unknown_args=False,
        json_file=None,
        json_file_encoding=None,
        yaml_file=None,
        yaml_file_encoding=None,
        toml_file=None,
        secrets_dir=None,
        protected_namespaces=('model_', 'settings_'),
    )


```